test/InternalFunctionsTest.php

```diff
@@ -9,16 +9,53 @@
 final class InternalFunctionsTest extends TestCase
 {
     /**
-     * Tests that the JSONB `@?` operator is not substituted by the parameter parser.
+     * Tests that unnamed parameters are substituted correctly.
+     *
+     * @see parseNamedParams
+     * @dataProvider provideUnnamedParams
+     */
+    public function testParseUnnamedParams(string $sqlIn, string $sqlOut): void
+    {
+        self::assertSame($sqlOut, parseNamedParams($sqlIn, $dummy));
+    }
+
+    public function provideUnnamedParams(): iterable
+    {
+        return [
+            'Bare' => ['SELECT ?', 'SELECT $1'],
+            'Parenthesized' => ['SELECT (?)', 'SELECT ($1)'],
+            'Row constructor' => ['SELECT (?, ?)', 'SELECT ($1, $2)'],
+        ];
```

Comment:
Hint: Yield could be used here, but I think there's no need yet for readability.